### ISSUE TITLE
Find a multilevel debug framework

### ISSUE THREAD

**Issue by kjetilk** [solid/acl-check#6]
This module uses mostly `console.log` and is pretty chatty. We need to use a debug framework with at least `info`, `debug` and `trace` levels. 

We should pick something that could be brought to v.next too.

**Comment by RubenVerborgh** [comment]
Note: NSS currently uses the `debug` module (used by Express as well).

**Comment by kjetilk** [comment]
Yeah, but that doesn't appear to have multiple levels, which I feel makes it unusable for this kind of project. Or does it?

**Comment by RubenVerborgh** [comment]
It does not, and it's a good counterargument.

**Comment by Ryuno-Ki** [comment]
Check out https://github.com/solid/node-solid-server/issues/920 - I suggested loglevel and winston there. loglevel is quite easy to use, whereas winston is quite sophisticated.

As starter I'd move the logging logic into a dedicated file, so you can exchange the logic there if needed (even if that module is just a proxy to console.log in the beginning).

**Comment by kjetilk** [comment]
@Ryuno-Ki do you know if loglevel is async?

I'd like to keep quite a lot of logging statements in, as I have found that it helps in projects like ours when people report problems, we can ask them to turn their log level up and try again. But then, it becomes really important that logging isn't occupying more than necessary of resources.

**Comment by Ryuno-Ki** [comment]
@kjetilk From reading the source code I'd say it is synchronous: https://github.com/pimterry/loglevel/blob/master/lib/loglevel.js
